### ISSUE TITLE
[INFO] Socket timed out / IP ban

### ISSUE THREAD

**Issue by Lautenschlager-id** [Lautenschlager-id/Transfromage#8]
Luvit sockets continue open even when the application is closed, and it takes a certain amount of time for it to disconnect.

That said, it's impossible to fix this problem unless you invest in stability:
- You should always call `client:closeAll()` before closing the application, since there's no way to auto-call it in unexpected problems, which may be handled by using `pcall` or `xpcall`.
```Lua
local timer = require("timer")

xpcall(function()
	-- Chunk
end, function(err)
	client:disconnect()
	-- Waits for disconnect to close everything and throws the error
	timer.setTimeout(1000, error, err)
end)
```
- Refresh your internet modem to ~~probably~~ get a new IP and bypass the ban.
- Host your bot in a VPS (or Heroku) to bypass this issue.

**Comment by Blank3495** [comment]
you should change `client:closeAll()` to `client:disconnect()` since the former is now deprecated

**Comment by Lautenschlager-id** [comment]
Solved in the commit [9aba89b](https://github.com/Lautenschlager-id/Transfromage/commit/9aba89b141f4e124f7ff5d490393036c5ef87140) (version 2.4.0)

Thanks @Minstfm
